code/src/main/java/habit_mode/view/codebehind/SudokuScreenCodeBehind.java

```diff
@@ -0,0 +1,225 @@
+        assert this.sudokuScreenLabel != null : "fx:id=\"sudokuScreenLabel\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sudokuButton != null : "fx:id=\"sudokuButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.oneButton != null : "fx:id=\"oneButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.twoButton != null : "fx:id=\"twoButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.threeButton != null : "fx:id=\"threeButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.fourButton != null : "fx:id=\"fourButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.fiveButton != null : "fx:id=\"fiveButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sixButton != null : "fx:id=\"sixButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sevenButton != null : "fx:id=\"sevenButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.eightButton != null : "fx:id=\"eightButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.nineButton != null : "fx:id=\"nineButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        this.addTextFields();
+        this.setPaneListener();
+
+    }
+
+    private void setPaneListener() {
+        this.mainPane.sceneProperty().addListener((obs, wasNull, exists) -> {
+            if (this.mainPane.sceneProperty().isNotNull().get()) {
+                ((ServerServerCommunicator) this.viewModel.getServerCommunicator()).setToken((String) this.mainPane.getScene().getRoot().getUserData());
+            }
+        });
+    }
+
+    void addTextFields() {
+        for (int row = 0; row < 9; row++) {
+            for (int column = 0; column < 9; column++) {
+                Pane pane = new Pane();
+                Label label = new Label();
+                label.alignmentProperty().set(Pos.CENTER);
+                label.setFont(Font.font(15));
+                label.textAlignmentProperty().set(TextAlignment.CENTER);
+                label.layoutXProperty().bind(pane.widthProperty().subtract(label.widthProperty()).divide(2));
+                label.layoutYProperty().bind(pane.heightProperty().subtract(label.heightProperty()).divide(2));
+                pane.getChildren().add(label);
+                pane.setBackground(new Background(new BackgroundFill(Color.web("#D3D3D3"), CornerRadii.EMPTY, Insets.EMPTY)));
+                pane.setStyle("-fx-border-color: black");
+                pane.setMinHeight(35);
+                pane.setMinWidth(35);
+                EventHandler<javafx.scene.input.MouseEvent> eventHandler = new EventHandler<javafx.scene.input.MouseEvent>() { 
```

Comment:
Import `javafx.scene.input.MouseEvent` at the top so you don't have to be so explicit down here.